superbench/runner/runner.py

```diff
@@ -173,6 +178,72 @@ def fetch_results(self):    # pragma: no cover
             )
         )
 
+    def __create_results_summary(self):    # pragma: no cover
+        """Create the result summary files."""
+        all_results = list()
+        for node_path in (self._output_path / 'nodes').glob('*'):
+            if not node_path.is_dir():
+                continue
+            results_summary = dict()
+            reduce_ops = dict()
+            file_list = natsorted([str(f) for f in node_path.glob('**/results.json')])
+            file_list = [Path(f) for f in file_list]
+            for results_file in file_list:
+                with results_file.open() as f:
+                    results = json.load(f)
+                    for result in results:
+                        if result['name'] not in results_summary:
+                            results_summary[result['name']] = defaultdict(list)
+                        for metric in result['result']:
+                            results_summary[result['name']][metric].append(result['result'][metric])
+                            reduce_ops[metric] = result['reduce_op'][metric]
```

Comment:
What happens if `result['reduce_op'][metric]` is different for different `result`. Do we need to check whether all are the same?

Reply:
Added check for inconsistent reduce type.

Reply:
Added checking for Inconsistent reduce type.